modules/local/jellyfish/main.nf

```diff
@@ -0,0 +1,98 @@
+process COUNT {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mer-jellyfish:2.3.1--h4ac6f70_0' :
+        'biocontainers/kmer-jellyfish:2.3.1--h4ac6f70_0' }"
+
+    input:
+        tuple val(meta), path(fasta)
+
+    output:
+        tuple val(meta), path("*.jf"), emit: kmers
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+        """
+        if [[ ${fasta} == *.gz ]]; then
+            zcat ${fasta} > ${fasta.baseName}.fasta
+        fi
+        if [[ ${fasta} == *.fa ]]; then
+            cp ${fasta} ${fasta.baseName}.fasta 
+        fi
+        if [[ ${fasta} == *.fastq ]]; then
+            cp ${fasta} ${fasta.baseName}.fasta 
+        fi
+        jellyfish count \\
+         -m ${params.kmer_length} \\
+         -s 140M \\
+         -C \\
+         -t $task.cpus ${fasta.baseName}.fasta 
+         
+        mv mer_counts.jf ${prefix}_mer_counts.jf
+        """
+}
+
+process HISTO {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mer-jellyfish:2.3.1--h4ac6f70_0' :
+        'biocontainers/kmer-jellyfish:2.3.1--h4ac6f70_0' }"
+
+    input:
+        tuple val(meta), path(kmers)
+
+    output:
+        tuple val(meta), path("*.tsv"), emit: histo
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+        """
+        jellyfish histo $kmers > ${prefix}_hist.tsv         
+        """
+}
+
+process STATS {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mer-jellyfish:2.3.1--h4ac6f70_0' :
+        'biocontainers/kmer-jellyfish:2.3.1--h4ac6f70_0' }"
+    input:
+        tuple val(meta), path(kmers)
+
+    output:
+        tuple val(meta), path("*.txt"), emit: stats
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+        """
+        jellyfish stats $kmers > ${prefix}_stats.txt       
```

Comment:
Will you swap to FastK / Meryl, or are you planning on sticking with Jellyfish?

Reply:
Meryl is used in the pipeline for merqury QC.
Jellyfish/genomescope part is _only_ to estimate genome size from ONT reads, which may be a) not necessary since flye no longer requires a genomesize and b) not very accurate since it involves some guessing about the read length. I am not sure if this is very important functionality.

Reply:
Genomescope is very important QC check of the data. Since ONT reads are getting more accurate, it's definitely better to have this functionality, although Genomescope can be run from Meryl/FastK databases too. Also note, there's GOAT ( Genomes on a tree ) which can be queried to get species information estimates like genome size, haploid number, etc. It's a good sanity check that the assembly size doesn't vary wildly from the estimated genome size from genomescope or GOAT, but there are reasons it might.